qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -0,0 +1,265 @@
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": True,
+                    "ZX plain": False,
+                    "Theta": new_theta
+                }
+        else:
+            new_qubit_state = qubit_state
+        # new_qubit_state = qubit_state
+        return new_qubit_state
+
+    def _rx_gate(self, qubit_state: dict, angle: float) -> dict:
+        """
+        Apply Rx gate.
+        Args:
+            qubit_state(dict): The state of the qubit before operating the gate.
+            angle(float): The angle of the rotation.
+
+        Returns:
+                dict: The state of the qubit after operating the gate.
+        """
+        if qubit_state["XY plain"]:
+            if isclose(angle, np.pi):
+                new_theta = - qubit_state["Theta"]
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": True,
+                    "ZX plain": False,
+                    "Theta": new_theta,
+                }
+            elif isclose(angle, np.pi/2):
+                new_theta = angle - qubit_state["Theta"]
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": False,
+                    "ZX plain": True,
+                    "Theta": new_theta,
+                }
+            else:
+                print("Error - This angle isn't supported. We only support multipication of pi/2")
+        else:
+            if isclose(angle, np.pi/2):
+                new_theta = qubit_state["Theta"] + 3 * np.pi/2  # its theta -pi/2 but we added 2*pi
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": True,
+                    "ZX plain": False,
+                    "Theta": new_theta,
+                }
+            elif isclose(angle, np.pi):
+                new_theta = qubit_state["Theta"] + np.pi
+                new_theta = new_theta % (2 * np.pi)
+                new_qubit_state = {
+                    "XY plain": False,
+                    "ZX plain": True,
+                    "Theta": new_theta,
+                }
+            else:
+                print("Error - This angle isn't supported. We only support multiplication of pi/2")
+        return new_qubit_state
+
+
+    def _measurement_gate(self, qubit_state: dict) -> int:
+        """
+        implementing measurement on qubit with read-out error.
+        Args:
+            qubit_state(dict): The state of the qubit at the end of the circuit.
+
+        Returns:
+                int: The result of the measurement after applying read-out error.
+        """
+        if qubit_state["XY plain"]:
+            meas_res = self._rng.random() < 0.5
+        else:
+            z_projection = np.cos(qubit_state["Theta"])
+            probability = abs(z_projection) ** 2
```

Comment:
`z_projection` is a real number, if you square it then you don't need `abs` in addition.

Reply:
Changed as suggested.